### ISSUE TITLE
recentf is loaded too late

### ISSUE THREAD

**Issue by marienz** [syl20bnr/spacemacs#7466]
#### Description :octocat:

`recentf` is loaded too late if `recents` is not in `dotspacemacs-startup-lists`.
#### Reproduction guide :beetle:
- Remove `recents` from `dotspacemacs-startup-lists` and use ivy for completion
- Start Emacs
- Press `SPC b b`

_Observed behaviour:_ :eyes: :broken_heart:
Recently opened files are not listed by `SPC b b` unless I find some file first, and if I wait 10 minutes without finding a file an error message about `recentf-save-list` appears in the minibuffer.

_Expected behaviour:_ :heart: :smile:
`SPC b b` works the same way immediately after starting Emacs as after triggering a load of recentf, and there is no error message.

The error message occurs because spacemacs installs an idle timer calling `recentf-save-list`, but that function is not autoloaded. So if the idle timer expires before something has triggered a load of `recentf`, the call fails. It should probably install that timer only after recentf has been loaded.

The inconsistent behavior from `SPC b b` occurs because spacemacs sets `ivy-use-virtual-buffers` to `t` only after `recentf` is loaded.

Finding some file first fixes it because spacemacs adds a function to `find-file-hook` that turns on `recentf-mode` if it is not yet enabled.

There is a comment in the ivy layer that claims "If we set this (`ivy-use-virtual-buffers`) before recentf loads, then ivy-mode loads recentf for us, which messes up the spacemacs version of recentf.". I do not understand how that would happen, since neither ivy nor spacemacs seems to be doing anything special to turn on recentf-mode (they just call `(recentf-mode)` or `(recentf-mode 1)`), and I do not see how I might trigger ivy before I can find a file (so I do not think ivy can cause recentf to load too soon). Maybe `ivy-use-virtual-buffers` can be set to `t` sooner to make `SPC b b` always show recent files?
#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.1.1
- Spacemacs: 0.200.2
- Spacemacs branch: master (rev. 7bdde17)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: ivy
- Layers:

``` elisp
(ivy auto-completion better-defaults emacs-lisp
     (go :variables go-tab-width 4)
     python c-c   markdown sql systemd shell-scripts git
     (shell :variables shell-default-shell 'eshell shell-default-height 30 shell-default-position 'bottom)
     syntax-checking version-control)
```

**Comment by TheBB** [comment]
This is the description of the commit that introduced that setting.

    commit a8c5119a50609fb6ac5d3bd3ad9cab3f6fab69e8
    Author: justbur <justin@burkett.cc>
    Date:   Wed Jan 6 10:51:54 2016 -0500

        ivy: Fix recentf loading

        ivy-use-virtual-buffers was triggering recentf-mode to be called before
        the recentf init function was called leading to problems with where the
        recentf file was placed.

So if ivy loads recentf before `(spacemacs-base/init-recentf)` has run, the value of `recentf-save-file` might be wrong. Now I don't know if or why ivy actually causes recentf to load but we could probably fix this by moving recentf to the pre step.

**Comment by marienz** [comment]
The Ivy part of this got fixed at some point, probably by 9b24a38eb95966f9a5fb74d7ba9d882e66e31146.

I didn't wait for the timer to expire to confirm it, but I see an autoload for `recentf-save-list` now, so I'm guessing the idle timer now also work.
